### PR TITLE
R&I Layout editing mods

### PR DESCRIPTION
**Pull request by charlesh88**
Changes focused on improving editing in Layouts with an emphasis on sub-object editing. 
**READY FOR REVIEW**

### Changes
- Much CSS cleanup and reorging for clarity;
- Removed `is-drilled-in` classes;
- Solidified styles for c-frame-edit*;
- Properly handle `.has-complex-content`;
- Grid visibility now working properly for nested layouts, but approach is brittle as hell;
- Removed `.l-layout__object` container, not needed;
- Fixed `--no-frame` hiding downstream nested `__header` elements;
- Added missing description for DisplayLayoutType;
- Add complex-content class implementation for Layout sub-objects with complex views, including 'move bar' with grippy affordance;
- Added `is-resizing` property to DisplayLayout.vue to detect when a user is actively resizing a Layout's frame, in order to prevent intermittent hovers from triggering;
- Set frame title by reading domain object name directly;
- Fixed border styling for Display and Flexible Layout frame elements while editing;
- Flex Layout headers and object Elements pool drag grippies now uses same style as Display Layouts;
- Toolbar now has background color as a further visual key into editing mode;

### Author Checklist
1. Changes address original issue? Y
1. Unit tests included and/or updated with changes? N/A
1. Command line build passes? Y
1. Changes have been smoke-tested? Y

### CODE REVIEW

**Comment by akhenry** [comment]
> ** IN PROGRESS **

This still in progress, or is it good to review?

File: src/ui/components/ObjectFrame.vue [hunk 1]
```diff
@@ -124,6 +130,17 @@
...
+            complexContent() {
+                if (!this.domainObject || !this.domainObject.type) {
+                    return;
+                }
+                let objectType = this.openmct.types.get(this.domainObject.type);
+                if (!objectType || !objectType.definition) {
+                    return; // TODO: return unknown icon type.
+                }
+
+                return objectType.definition.complexContent;
```
[the remark this answers is not in the document]
Comment (charlesh88) [reply src/ui/components/ObjectFrame.vue]: @dtailor90 to address.

[the remark this answers is not in the document]
Comment (charlesh88) [reply src/ui/components/ObjectFrame.vue]: Merged latest TCR, this should be good.

File: src/plugins/displayLayout/components/DisplayLayout.vue [hunk 2]
```diff
@@ -139,22 +151,6 @@
             addElement(itemType, element) {
                 this.addItem(itemType + '-view', element);
             },
-            setSelection(selection) {
-                if (selection.length === 0) {
-                    return;
-                }
-
```
[the remark this answers is not in the document]
Comment (charlesh88) [reply src/plugins/displayLayout/components/DisplayLayout.vue]: Merged latest TCR, this should be good.

File: src/api/types/TypeRegistry.js [hunk 3]
```diff
@@ -77,6 +78,28 @@ define(['./Type'], function (Type) {
         }
     };
 
+    /**
+     * Check if type is included in complex types.
+     * @private
+     */
+    TypeRegistry.prototype.isTypeComplex = function (key) {
```
[the remark this answers is not in the document]
Comment (charlesh88) [reply src/api/types/TypeRegistry.js]: Per discussion with @akhenry and @dtailor90:
- List of complex content objects to be opt-out, and maintained within the Layout plugin. 
- @dtailor90 to address.

File: src/plugins/displayLayout/components/DisplayLayout.vue [hunk 4]
```diff
@@ -234,12 +230,17 @@
                 this.initSelectIndex = this.layoutItems.length - 1;
             },
             trackItem(item) {
+                if (!item.identifier) {
```
[the remark this answers is not in the document]
Comment (akhenry) [reply src/plugins/displayLayout/components/DisplayLayout.vue]: Got it, thanks

**Comment by akhenry** [comment]
Changes look good to me. Thanks!

**Pull request merged**